### PR TITLE
checker: speed up merge regions by batch

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve? 

The PR will try to solve #1430 .
- [x]  Add operators in batches when patrolling regions.
- [x]  Bench
- [x]  Test Report

For merge empty region, it speed up 10.8-14.57 times.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by nolouch** [comment]
Maybe not only the merge checker but all checkers can also speed up.

File: server/coordinator.go [hunk 1]
```diff
@@ -155,41 +159,37 @@ func (c *coordinator) drivePushOperator() {
...
+func (c *coordinator) checkRegion(region *core.RegionInfo) (bool, []*operator.Operator) {
 	opController := c.opController
-
+	checkerIsBusy := true
 	if op := c.learnerChecker.Check(region); op != nil {
-		if opController.AddOperator(op) {
-			return true
-		}
+		checkerIsBusy = false
+		return checkerIsBusy, []*operator.Operator{op}
```
Comment (disksing) [inline server/coordinator.go]: I think the local variable `checkerIsBusy` makes less readable. I prefer `return false, []Operator{op}` directly.

File: server/coordinator_test.go [hunk 2]
```diff
@@ -276,6 +276,15 @@ func (s *testCoordinatorSuite) TestCollectMetrics(c *C) {
 		co.cluster.collectClusterMetrics()
 	}
 }
+func MaxUint64(nums ...uint64) uint64 {
```
Comment (disksing) [inline server/coordinator_test.go]: add a space line before it.

File: server/coordinator.go (hunk 1 above)
Comment (lhy1024) [reply server/coordinator.go]: It is acceptable for `learnerChecker`.But for other checker, there is one case that checker is not busy and region cannot through checker.

**Comment by nolouch** [comment]
How about the benchmark result?

**Comment by lhy1024** [comment]
> How about the benchmark result? @nolouch 

Bench is raw without real schedule, I am trying to test it on real cluster.
```
BenchmarkPatrolRegion-12               1        3308208403 ns/op        27595632 B/op     714200 allocs/op
BenchmarkPatrolRegion-12               1        205220668889 ns/op      40874184 B/op     981827 allocs/op
```
It has reached an astonishing 62 times, but in fact it should be only 12-16 times

**Comment by lhy1024** [comment]
master branch

image
image

Case | finish spilt | fast merge | finish merge |first stage | second stage | total
-- | -- | -- | -- | -- | -- | --
1 | 22：30 | 23：14 | 02：10 | 44 min | 2h56min | 3h40min
2 | 18：00 | 19：09 | 21：30 | 1h09min | 2h21min | 3h30min
3 | 11：28 | 12：18 | 14：57 | 50min | 2h39min | 3h29min
average | - | - | - | 55min | 2h38min | 3h33min

speed-up branch
image
image

Case | start merge | finish merge | total
-- | -- | -- | --
4 | 16：30 | 16：47 | 17min
5 | 14：31 | 14：46 | 15min
6 | 21：17 | 21：41 | 24min
average | - | - | 19min

Only the second stage of statistics is achieved, there is an optimization of 8.3 times. Considering two stages, the optimization effect is 11.2 times.

**Pull request merged**